packages/graphql-language-service-server/src/GraphQLCache.ts

```diff
@@ -296,11 +292,7 @@ export class GraphQLCache implements GraphQLCacheInterface {
     }
     const filesFromInputDirs = await this._readFilesFromInputDirs(
```

Comment:
@ardatan i had some questions about `await project.loadDocuments()`, as I think it could replace a lot of functionality we have in `GraphQLCache`. Our implementation has a different structure, where there is a collection item per file, and thus potentially multiple asts and other extracted data, whereas with `Source[]` it seems to be a collection of ASTs/strings with potentially entries with the same `location` file reference. I think we can at least get rid of our redundant logic to glob, read and parse files seperately here. There may be a bit of data missing for this to be accomplished, for which I can open a PR to `graphql-config` to add that data to these interfaces?

Reply:
@acao Sure! You can open a PR for that :)

Reply:
Looking into this further and I don't know if I'll need to actually!

paired with this PR, all of this is working:
https://github.com/prisma-labs/vscode-graphql/pull/184/files

however, there is some redunant logic between our GraphQLCache and MessageProcessor now, as of this PR, and I think I need to rewrite some of this. Either way, going to see if making it more `graphql-config` based in file loading will help! will definately make it more extensible!

Reply:
@ardatan the way that we cache some of the information about files, we need to store the range. it would be nice if there was a `source.range` in `getDocuments()` from which to compute offsets

Reply:
it seems i can address this issue with custom loader extensions as well. i need to be able to set specific parser configs as well. probably will be the solution in the long run